torchgeo/datamodules/utils.py

```diff
@@ -52,3 +54,75 @@ def dataset_split(
             [train_length, val_length, test_length],
             generator=Generator().manual_seed(0),
         )
+
+
+def group_shuffle_split(
+    groups: Any,
+    train_size: Optional[float] = None,
+    test_size: Optional[float] = None,
+    random_state: Optional[int] = None,
+) -> tuple[Any, Any]:
+    """Method for performing a single group-wise shuffle split of data.
+
+    Loosely based off of `sklearn.model_selection.GroupShuffleSplit`.
+
+    Args:
+        groups: a sequence of group values used to split. Should be in the same order as
+            the data you want to split.
+        train_size: the proportion of groups to include in the train split. If None,
+            then it is set to complement `test_size`.
+        test_size: the proportion of groups to include in the test split (rounded up).
+            If None, then it is set to complement `train_size`.
```

Comment:
Is there any reason to have both params if train_size + test_size = 1?

Reply:
so you can specify either train size or test size

Reply:
Could instead support a list of lengths/fractions to match `torch.utils.data.random_split` and `torchgeo.datasets.splits`. This would support train/val/test splits in addition to train/test splits.